traitsui/qt4/table_editor.py

```diff
@@ -594,6 +559,53 @@ def _get_image(self, image):
 
         return self.images.get(image)
 
+    def _create_empty_menu(self):
+        """Create a QMenu to display in empty space below the rows.
+
+        Returns a QMenu or None if no menu to display.
+        """
+        if not self.factory.editable or self.factory.row_factory is None:
+            return None
+
+        empty_menu = QtGui.QMenu(self.table_view)
+        action = empty_menu.addAction("Add new item")
+        action.triggered.connect(self._on_context_append)
+        return empty_menu
+
+    def _create_header_menu(self):
+        """Create a QMenu to display in the vertical header.
+
+        Returns a QMenu or None if no menu to display.
+        """
+        header_menu = QtGui.QMenu(self.table_view)
+        insertable = self.factory.row_factory is not None
+        editable = self.factory.editable
+        deletable = self.factory.deletable
+        reorderable = self.factory.reorderable
```

Comment:
Not sure how I feel about these given they're only referenced once.

Reply:
I think these are there because the code originally came from the deleted code above, like this:
```
if factory.editable and (insertable or factory.deletable):
```
but with added `self.`.  But then I refactored.  It probably makes sense to remove them.